### DISCUSSION TITLE
Is importing by the absolute specifier stable?

### DISCUSSION THREAD

**Discussion by Kapelianovych** [nodejs/help#4962] [Q&A] [answered]
Hi folks!

I'm trying to import CommonJS scripts by the absolute specifier.

```js
const fileId = pathToFileURL(path.resolve('rootFile.js')).toString();

const module = require(fileId);
```

That code throws an error:

```
Error: Cannot find module 'file:///absolute/path/to/package/rootFile.js'
```

But I'm sure that it is there and the path is correct. The fun thing is that the following code works just fine:

```js
const fileId = pathToFileURL(path.resolve('rootFile.mjs')).toString();

const module = await import(fileId);
```

And the next too:

```js
const fileId = path.resolve('rootFile.js');

const module = require(fileId);
```

I'm using MacOS 11.6.7 with Node 16.16.0.

Maybe it's worth mentioning that I'm trying to import the file from the other package. Though it's *package.json* doesn't have neither the `main` files, nor the `exports` one.

What's going on? Is it the bug in NodeJS or I doesn't know something (which is likely)?

**Comment by ichenglin** [comment, marked as answer]
**`pathToFileURL().toString();`** converts the path into **url-encoded file protocol string**,
and **[node require](https://nodejs.org/en/knowledge/getting-started/what-is-require/)** simply **doesn't support** using that format to import modules.

**Reply by Kapelianovych** [reply]
Thanks for the answer!
